File: Cap xB/calculadoraV4.dart
import 'dart:io';

List<double> entrada() {
  stdout.write("Entre o 1o. valor: ");
  var input = stdin.readLineSync();
  var a = double.parse(input!);
  stdout.write("Entre o 2o. valor: ");
  input = stdin.readLineSync();
  var b = double.parse(input!);
  return [a, b];
}

void saida(double r) {
  print("\nO resultado da operacao equivale a: ${r.toStringAsFixed(2)}.\n");
}

double calculo(double a, double b, String operador) {
  switch (operador) {
    case '+':
      return a + b;
    case '-':
      return a - b;
    case '*':
      return a * b;
    case '/':
      return a / b;
    default:
      return 0;
  }
}

void rotcalc(String operacao) {
  switch (operacao) {
    case '+':
      print("\nRotina de Adicao\n");
      break;
    case '-':
      print("\nRotina de Subtracao\n");
      break;
    case '*':
      print("\nRotina de Multiplicacao\n");
      break;
    case '/':
      print("\nRotina de Divisao\n");
      break;
  }
  var valores = entrada();
  var a = valores[0];
  var b = valores[1];  
  if (operacao == '/') {
    if (b == 0) {
      print("\nO resultado da operacao equivale a: ERRO.\n");
    } else {
      var r = calculo(a, b, '/');
      saida(r);
    }
  }
  if (operacao != '/') {
    var r = calculo(a, b, operacao);
    saida(r);
  }
}

void main() {
  var opcao = 0;
  while (opcao != 5) {
    print("CALCULADORA - V4");
    print("================\n");
    print("1 - Adicao");
    print("2 - Subtracao");
    print("3 - Multiplicacao");
    print("4 - Divisao");
    print("5 - Fim de Programa\n");
    stdout.write("Escolha uma opcao: ");
    var input = stdin.readLineSync();
    opcao = int.parse(input!);
    if (opcao != 5) {
      switch (opcao) {
        case 1:
          rotcalc('+');
          break;
        case 2:
          rotcalc('-');
          break;
        case 3:
          rotcalc('*');
          break;
        case 4:
          rotcalc('/');
          break;
        default:
          print("\nOpcao invalida - Tente novamente.\n");
          break;
      }
    }
  }
}

File: Cap xB/calculadoraV4.ts
const readline = require('readline-sync');

function ENTRADA() {
  process.stdout.write('Entre o 1o. valor: ');
  A = parseFloat(readline.prompt());
  process.stdout.write('Entre o 2o. valor: ');
  B = parseFloat(readline.prompt());
}

function SAIDA() {
  process.stdout.write(`\nO resultado da operacao equivale a: ${R.toFixed(2)}.\n\n`);
}

function CALCULO(X, Y, OPERADOR) {
  let resultado = 0;
  switch (OPERADOR) {
    case '+':
      resultado = X + Y;
      break;
    case '-':
      resultado = X - Y;
      break;
    case '*':
      resultado = X * Y;
      break;
    case '/':
      resultado = X / Y;
      break;
  }
  return resultado;
}

function ROTCALC(OPERAÇÃO) {
  switch (OPERAÇÃO) {
    case '+':
      console.log('\nRotina de Adicao\n');
      break;
    case '-':
      console.log('\nRotina de Subtracao\n');
      break;
    case '*':
      console.log('\nRotina de Multiplicacao\n');
      break;
    case '/':
      console.log('\nRotina de Divisao\n');
      break;
  }
  ENTRADA();
  if (OPERAÇÃO === '/') {
    if (B === 0) {
      process.stdout.write('\nO resultado da operacao equivale a: ERRO.\n\n');
    } else {
      R = CALCULO(A, B, '/');
      SAIDA();
    }
  } else {
    R = CALCULO(A, B, OPERAÇÃO);
    SAIDA();
  }
}

import readlineSync from 'readline-sync';

let A: number, B: number, R: number;

function ENTRADA(): void {
  A = parseFloat(readlineSync.question('Entre o 1o. valor: '));
  B = parseFloat(readlineSync.question('Entre o 2o. valor: '));
}

function SAIDA(): void {
  console.log(`\nO resultado da operacao equivale a: ${R.toFixed(2)}.\n`);
}

function CALCULO(X: number, Y: number, OPERADOR: string): number {
  let resultado = 0;
  switch (OPERADOR) {
    case '+':
      resultado = X + Y;
      break;
    case '-':
      resultado = X - Y;
      break;
    case '*':
      resultado = X * Y;
      break;
    case '/':
      resultado = X / Y;
      break;
  }
  return resultado;
}

function ROTCALC(OPCAO: string): void {
  switch (OPCAO) {
    case '+':
      console.log('\nRotina de Adicao\n');
      break;
    case '-':
      console.log('\nRotina de Subtracao\n');
      break;
    case '*':
      console.log('\nRotina de Multiplicacao\n');
      break;
    case '/':
      console.log('\nRotina de Divisao\n');
      break;
  }
  ENTRADA();
  if (OPCAO === '/') {
    if (B === 0) {
      console.log('\nO resultado da operacao equivale a: ERRO.\n');
    } else {
      R = CALCULO(A, B, '/');
      SAIDA();
    }
  } else {
    R = CALCULO(A, B, OPCAO);
    SAIDA();
  }
}

let OPCAO = '0';

while (OPCAO !== '5') {
  console.log('CALCULADORA - V4');
  console.log('================\n');
  console.log('1 – Soma');
  console.log('2 – Subtracao');
  console.log('3 – Multiplicacao');
  console.log('4 – Divisao');
  console.log('5 – Fim de Programa\n');
  OPCAO = readlineSync.question('Escolha uma opcao: ');
  if (OPCAO !== '5') {
    switch (OPCAO) {
      case '1':
        ROTCALC('+');
        break;
      case '2':
        ROTCALC('-');
        break;
      case '3':
        ROTCALC('*');
        break;
      case '4':
        ROTCALC('/');
        break;
      default:
        console.log('\nOpcao invalida - Tente novamente.\n');
    }
  }
}
